pkg/util/immutable/immutable.go

```diff
@@ -98,7 +98,12 @@ func validate(path string, v, w reflect.Value, ignoreCase bool) error {
 		for i.Next() {
 			k := i.Key()
 
-			err := validate(fmt.Sprintf("%s[%q]", path, k.Interface()), v.MapIndex(k), w.MapIndex(k), ignoreCase)
+			mapV, mapW := v.MapIndex(k), w.MapIndex(k)
+			if mapV.IsValid() != mapW.IsValid() {
```

Comment:
presumably mapV.IsValid() is guaranteed to be true given that we are iterating v?  Is something like `if !mapW.IsValid() { // corresponding key does not exist in w` clearer?

Reply:
Good point. I'll update the if